content/techniques/authentication.md

```diff
@@ -782,6 +816,81 @@ import { JwtStrategy } from './jwt.strategy';
 export class AuthModule {}
 ```
 
+#### Request-scoped strategies
+
+The passport API is based on registering strategies to the global instance of the library. Therefore strategies are not designed to have request-dependent options or to be dynamically instantiated per request (read more about the [request-scoped](/fundamentals/injection-scopes) providers). When you configure your strategy to be request-scoped, Nest will never instantiate it since it's not tied to any specific route. There is no physical way to determine which "request-scoped" strategies should be executed per request.
+
+However, there are ways to dynamically resolve request-scoped providers within the strategy. For this, we leverage the [module reference](/fundamentals/module-ref) feature.
+
+First, open the `local.strategy.ts` file and inject the `ModuleRef` using a normal way:
+
+```typescript
+constructor(private readonly moduleRef: ModuleRef) {
+  super({
+    passReqToCallback: true,
+  });
+}
+```
+
+> info **Hint** The `ModuleRef` class is imported from the `@nestjs/core` package.
+
+Next, make sure to set the `passReqToCallback` configuration property to `true`.
+
+```typescript
+super({
+  passReqToCallback: true,
+});
+```
+
+Request instance is required to obtain the current context identifier instead of generating a new one.
```

Comment:
Request instance is required to obtain the current context identifier instead of generating a new one.
->
In the next step, the request instance will be used to obtain the current context identifier, instead of generating a new one (read more about request context [here](https://docs.nestjs..com/fundamentals/module-ref#getting-current-sub-tree)).